src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/SqlQueryGenerator.cs

```diff
@@ -390,7 +394,8 @@ private void HandleTableKindTop(SearchOptions context)
             var sortExpression = (paramInfo == null || paramInfo.Code == KnownQueryParameterNames.LastUpdated) ? null : $"{tableExpressionName}.SortValue";
 
             // Everything in the top expression is considered a match
-            StringBuilder.Append("SELECT DISTINCT TOP (").Append(Parameters.AddParameter(context.MaxItemCount + 1)).Append(") Sid1, 1 AS IsMatch, 0 AS IsPartial ")
+            string selectTop = sortExpression == null ? "SELECT DISTINCT" : "SELECT";
```

Comment:
nit: maybe "Top" in the name is misleading since the TOP comes in the next line :)

Reply:
Updated name to `selectStatement` instead of `selectTop`